crates/matrix-sdk-indexeddb/src/crypto_store/mod.rs

```diff
@@ -713,9 +726,13 @@ impl_crypto_store! {
 
         tx.await.into_result()?;
 
-        // all good, let's update our caches:indexeddb
-        for session in changes.sessions {
-            self.session_cache.add(session).await;
+        if clear_caches {
+            self.clear_caches().await;
+        } else {
```

Comment:
Should this be inside an `else`? I'd expect to want to add the sessions to the cache even if we've just cleared the cache. I expect I'm misunderstanding what this is doing.

Reply:
We don't know if the sessions were created using the old or the new device keys, so we don't want to cache them.  (In fact, they're probably created using the old device keys, since we hadn't saved the new device keys yet).

Reply:
Makes sense - please could you add a comment explaining this?